src/views/reset-password/reset-password.tsx

```diff
@@ -0,0 +1,86 @@
+import { Button, Card, Help, InputField, LogoIcon } from '@dhis2/ui'
+import { useCallback, useState } from 'react'
+import {useNavigate, useParams} from 'react-router-dom'
+import { useAuthAxios } from '../../hooks'
+import styles from './reset-password.module.css'
+import {useAlert} from "@dhis2/app-service-alerts";
+
+const getInputsErrorMessage = ({ password, confirmPassword }) => {
+    if (!password) {
+        return 'Please provide a password'
+    } else if (!confirmPassword) {
+        return 'Please confirm your password'
+    } else if (password !== confirmPassword) {
+        return 'Passwords do not match'
+    } else {
+        return ''
+    }
+}
+
+export const ResetPassword = () => {
+    const navigate = useNavigate()
+    const { token } = useParams()
+    const [inputs, setInputs] = useState({
+        password: '',
+        confirmPassword: '',
+    })
+    const [errorMessage, setErrorMessage] = useState('')
+    const { show: showAlert } = useAlert(
+        ({ message }) => message,
+        ({ isCritical }) => (isCritical ? { critical: true } : { success: true })
+    )
+    const onInputChange = useCallback(({ value, name }) => {
+        // Always clear the error when user starts typing again
+        setErrorMessage('')
+        setInputs((currentInputs) => ({
+            ...currentInputs,
+            [name]: value,
+        }))
+    }, [])
+    const [{ loading }, postResetPassword] = useAuthAxios(
+        {
+            url: '/users/reset-password',
+            method: 'POST',
+            // data: { token, password },
+        },
+        { manual: true }
+    )
+
+    const onSubmit = useCallback(
+        async (event) => {
+            event.preventDefault()
+
+            if (inputs.password) {
```

Comment:
I think this whole section would be simpler if you reverse the if. The indentation level would be lower if you start by doing if (!inputs.password) { setErrorMessage(getInputsErrorMessage(inputs)); return }

Please note that this isn't something I feel super strongly about but I do consider it a minor improvement when it comes to readability :slightly_smiling_face: